themes/default/content/docs/pulumi-cloud/oidc/client/kubernetes-gke.md

```diff
@@ -0,0 +1,133 @@
+---
+title_tag: Configure OpenID Connect for Google Kubernetes Engine | OIDC
+meta_desc: This page describes how to configure Pulumi to accept Google Kubernetes Engine OIDC tokens
+title: Google Kubernetes Engine
+h1: Configuring OpenID Connect for Google Kubernetes Engine
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+    pulumicloud:
+        parent: openid-connect-client
+        weight: 1
+---
+
+This document outlines the steps required to configure Pulumi to accept Google Kubernetes Engine id_tokens to be exchanged by Organization access tokens
+
+## Prerequisites
+
+* You must be an admin of your Pulumi organization.
+
+{{< notes type="warning" >}}
+Please note that this guide provides step-by-step instructions based on the official provider documentation which is subject to change. For the most current and precise information, always refer to the [official Bound Service Account Tokens documentation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1205-bound-service-account-tokens).
+{{< /notes >}}
+
+## Register the OIDC issuer
+
+1. Navigate to **OIDC Issuers** under your Organization's **Settings** and click on **Register a new issuer**.
+1. Name the issuer and complete the url: `https://container.googleapis.com/v1/projects/PROJECT_NAME/locations/us-west1/clusters/CLUSTER_NAME/`
+   ![Register GKE](../register-gke.png)
+1. Submit the form
+
+## Conrigure the Authorization Policies
+
+1. Click on the issuer name
+1. Change the policy decision to `Allow`
+1. Change the token type to `Organization`
+1. Add a new rule and configure it to verify the pod requesting the token: `"kubernetes.io".pod.name: runner-*`.
+   ![kubernetes policy example](../kubernetes-policy.png)
+1. Click on update
+
+## Sample
+
+```typescript
+// Pulumi program to run a bash script in a Kubernetes pod,
+// mount a service account token with an appropriate audience,
+// exchange the token for an ordinary Pulumi access token,
+// then run `pulumi whoami`.
+
+import * as kubernetes from "@pulumi/kubernetes";
+
+const tokenParams = {
+    "audience": "urn:pulumi:org:ORG_NAME",
+    "token_type": "urn:pulumi:token-type:access_token:organization",
+    "expiration": 2 * 60 * 60,
+}
+
+const script = new kubernetes.core.v1.ConfigMap("script", {
```

Comment:
I assumed that we would provide a CLI command for this, or enhance the login command, to do the low-level token exchange on their behalf. Is there a design doc section where we could discuss further?

Seems almost premature to talk about this feature until we have this aspect sorted out.

Reply:
Eventually, we have to scope to the time we have available to ship the feature. There are tons of satellite work needed to support every use-case possible, I believe for the time being, it is a good tradeoff to provide guidelines on how this can be done and in the future replace with proper tooling support.